client/rpc/client.go

```diff
@@ -0,0 +1,164 @@
+
+	objReq := directfund.NewObjectiveRequest(
+		counterparty,
+		100,
+		outcome,
+		uint64(rand.Float64()), // TODO: Since numeric fields get converted to a float64 in transit we need to prevent overflow
+		common.Address{})
+
+	// Create a channel and store it in the responses map
+	// We will use this channel to wait for the response
+	resRec := make(chan interface{})
+	rc.responses.Store(string(objReq.Id(rc.myAddress, rc.chainId)), resRec)
+
+	requestId := rand.Uint64()
+	rc.idsToMethods.Store(fmt.Sprintf("%d", requestId), serde.DirectFundRequestMethod)
+
+	message := serde.NewJsonRpcRequest(requestId, serde.DirectFundRequestMethod, objReq)
+	data, err := json.Marshal(message)
+	if err != nil {
+		panic("Could not marshal direct fund request")
+	}
+	rc.nts.SendMessage(serde.DirectFundRequestMethod, data)
+
+	objRes := <-resRec
+	return objRes.(directfund.ObjectiveResponse)
+}
+
+func (rc *RpcClient) CloseLedger(id types.Destination) protocols.ObjectiveId {
+	objReq := directdefund.NewObjectiveRequest(id)
+
+	// Create a channel and store it in the responses map
+	// We will use this channel to wait for the response
+	resRec := make(chan interface{})
+	rc.responses.Store(string(objReq.Id(rc.myAddress, rc.chainId)), resRec)
+	requestId := rand.Uint64()
+	rc.idsToMethods.Store(fmt.Sprintf("%d", requestId), serde.DirectDefundRequestMethod)
+
+	message := serde.NewJsonRpcRequest(requestId, serde.DirectDefundRequestMethod, objReq)
+	data, err := json.Marshal(message)
+	if err != nil {
+		panic("Could not marshal direct fund request")
+	}
+	rc.nts.SendMessage(serde.DirectDefundRequestMethod, data)
+
+	objRes := <-resRec
+	return objRes.(protocols.ObjectiveId)
+}
+
+func (rc *RpcClient) Close() {
+	rc.nts.Close()
+}
+
+// registerHandlers registers error and response handles for the rpc client
+func (rc *RpcClient) registerHandlers() {
+
+	rc.nts.RegisterErrorHandler(func(id uint64, data []byte) {
+		panic(fmt.Sprintf("Objective failed: %v", data))
+	})
+
+	rc.nts.RegisterResponseHandler(func(id uint64, data []byte) {
+		rc.nts.Logger.Trace().Msgf("Rpc client received response: %+v", data)
+		method, reqFound := rc.idsToMethods.Load(fmt.Sprintf("%d", id))
+		if !reqFound {
+			panic(fmt.Sprint("Could not find request for response with id %D", id))
+		}
+
+		switch method {
+		case serde.DirectFundRequestMethod:
+			handleResponse[directfund.ObjectiveResponse](rc, data)
+		case serde.DirectDefundRequestMethod:
+			handleResponse[protocols.ObjectiveId](rc, data)
+
+		}
+	})
+}
+
+// handleResponse handles a response from the rpc server for the given client
+// It is not a member of the RpcClient so it can take advantage of generics
+func handleResponse[T serde.ResponsePayload](rc *RpcClient, data []byte) {
+	rpcResponse := serde.JsonRpcResponse[T]{}
+	err := json.Unmarshal(data, &rpcResponse)
+	if err != nil {
+		panic("could not unmarshal direct defund objective response")
```

Comment:
I don't think we know it is a direct defund objective response at this point?

Reply:
Good catch! https://github.com/statechannels/go-nitro/pull/1073/commits/7250a3242ac1b97ad503349f523c8e7d044b2f7f